docs/DeveloperGuide.md

```diff
@@ -1,38 +1,308 @@
 # Developer Guide
 
-## Acknowledgements
+## Table of contents
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+### 1. [Design & Implementation](#1-design--implementation)
+1.1 [Database](#11-database)\
+1.2 [User Storage](#12-user-storage)\
+1.3 [Timetable](#13-timetable)\
+1.4 [Delete History](#14-delete-history)\
+1.5 [Ui](#15-ui)\
+1.6 [Commands](#16-commands)\
+1.7 [User Module Mapping](#17-usermodulemappinglist)\
+1.8 [User University List Manager](#18-useruniversitylistmanager)
+### 2. [Product Scope](#2-product-scope)
+2.1 [Target User Profile](#21-target-user-profile)\
+2.2 [Value Proposition](#22-value-proposition)\
+2.3 [Problem Addressed](#23-problem-addressed)
+### 3. [Non-functional Requirements](#3-non-functional-requirements)
+### 4. [Glossary](#4-glossary)
+### 5. [Manual Testing](#5-instructions-for-manual-testing)
 
-## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## 1. Design & implementation
 
-## Product scope
-### Target user profile
+### 1.1 Database
 
-{Describe the target user profile}
+The database reads in SEP module data from `data.csv` and stores the useful data into an ArrayList of module mappings and universities.
 
-### Value proposition
+Upon starting easySEP, the DatabaseStorage will load each line from `data.csv`, parse the line using DatabaseParser, and store the data into the Database.
 
-{Describe the value proposition: what problem does it solve?}
+Relevant exceptions are thrown when there are unexpected scenarios. For instance, if `data.csv` cannot be found at the given file path, a FileNotFoundException is thrown.
+
+The following diagram illustrates the relationships between the three main database classes - DatabaseStorage, DatabaseParser, and Database.
+
+![Database Class Diagram](./images/Database_Class.png)
+
+The following diagram illustrates the flow of the program, from the initial loadDatabase call to the eventual completion of updating the entire database.
+
+![Database Sequence Diagram](./images/Database_Sequence.png)
```

Comment:
![image](https://user-images.githubusercontent.com/83072470/198494867-e2037234-e962-4192-b28b-5079d0463c6d.png)

Is the activation bar suppose for DatabaseStorage suppose to start?